### PR TITLE
MDBOOK: How to contribute text

### LINKED ISSUE #241: Article: How to Contribute to the Book
There should be an article on what is involved in making a contribution to the book. Whether that's a whole article, or just a grammar correction. Also, writing guidelines

Related to this is the idea that every article should link in its footer back to the markdown source on github for it, for easy editing

This should mention licensing information.

### PR DESCRIPTION
**Pull request by emalinus**
First go at "how to contribute" article

closes #241 
and #244

### CODE REVIEW

**Comment by Connoropolous** [comment]
Instead of adding 'Issue 241' in the title of the issue (use 'Edit' to rename), use #241 in the pull request description field. (Which you can also edit, by using the '...' dropdown menu by it.

File: doc/holochain_101/src/how_to_contribute.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+1) Log on to your GitHub account
+
+2) In the Holochain Rust repo, everything is under the `doc/holochain_101/src` folder. All markdown files are there, and some are nested in subfolders. Navigate to the following link: [https://github.com/holochain/holochain-rust/tree/develop/doc/holochain_101/src](https://github.com/holochain/holochain-rust/tree/develop/doc/holochain_101/src)
+
+3) Determine whether you are making, editing, or reviewing an article.
+
+### Access Rights
+
+Please note that while those with direct write access to the repository will be able to submit change requests without creating a fork, those without will have to create a fork. However, this actually adds very little complexity in practice, just meaning that those with write access to the repository will have to review and merge your pull request.
```
Comment (thedavidmeister) [inline doc/holochain_101/src/how_to_contribute.md]: this is a bit wordy. proposal:

> If you don't have write access to the repository you need to create a fork to contribute. Forking is easy. Click the "Fork" button in the top right hand corner of the Github UI.
>
> Github maintains a [comprehensive forking guide](https://help.github.com/articles/fork-a-repo/).

File: doc/holochain_101/src/how_to_contribute.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+4) Write the rest of your text, checking the "Preview" tab to see how it would look.
+
+5) Scroll to the bottom of the page and select the option "create a new branch for this commit and start a pull request". You can name a branch, though GitHub will set one automatically. If you know it, mention the issue that the request addresses.
+
+6) Click "Propose New File". Proceed to the Opening a Pull Request section.
+
+### Editing an article you created
+
+1) Navigate to the article you want to edit, making sure you are on the branch where it was first created. Check which branch you are on by looking for the "Branch: ?" dropdown. Use the dropdown to switch to your branch if you're on the wrong one.
```
Comment (thedavidmeister) [inline doc/holochain_101/src/how_to_contribute.md]: is this right? i expected that after an article is merged we create new branches for new updates

File: doc/holochain_101/src/how_to_contribute.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+1) Under the Pull Request tab, look for ones starting with "MDBOOK". Go to the Pull Request of your choice, and then click on the "Files Changed" tab.
+
+2) Start a review by hovering over a line and pressing the blue "add" symbol to add comments to a line
+
+3) Click the green "Review Changes" button. If you approve of the changes, select "Approve". If you would like further changes to be made before it gets merged, select "Request Changes". If you are just weighing in, select "Comment". Then, click "Submit Review".
+
+### Merging a Pull Request
+
+3) Under "Conversation" you can merge the pull request, which integrates it into the `develop` branch. Changes are automatically deployed to [https://holochain.github.io/holochain-rust](https://holochain.github.io/holochain-rust) and usually appear there within 30 minutes. Merging a pull request should only be performed after it has received two "Approval" reviews.
```
Comment (thedavidmeister) [inline doc/holochain_101/src/how_to_contribute.md]: run on sentence. proposal:

> Changes automatically deploy to [https://holochain.github.io/holochain-rust](https://holochain.github.io/holochain-rust) within 30 minutes.

Comment (thedavidmeister) [reply doc/holochain_101/src/how_to_contribute.md]: passive voice. proposal:

> Aim for two approved reviews before merging.

File: doc/holochain_101/src/how_to_contribute.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
+# How to contribute
+
+This book uses a tool that builds HTML files from markdown files called ['mdbook'](https://github.com/rust-lang-nursery/mdBook). The markdown files are stored on GitHub, in the main `holochain-rust` repository. Because they are on GitHub, they have built-in version control, meaning that it's easy for anyone to contribute, and to propose, track and merge changes.
+
+There are two main pathways to contribute. One is by editing files directly in GitHub, and the other is cloning the repository to your computer, running mdbook locally, and then submitting change requests. The following covers both scenarios.
+
+## Writing Guidelines
+
+Please do not edit the SUMMARY.md file, which stores the raw chapter structure, without advanced signoff from the documentation team.
```
Comment (thedavidmeister) [inline doc/holochain_101/src/how_to_contribute.md]: link to the docs channel in mattermost?

File: doc/holochain_101/src/how_to_contribute.md (hunk 2 above)
Comment (emalinus) [reply doc/holochain_101/src/how_to_contribute.md]: This is how Connor explained the process...it may make more sense to make new branches for new updates once the contributions are coming more from outside the team, though.

Comment (Connoropolous) [reply doc/holochain_101/src/how_to_contribute.md]: We could divvy up these instructions a little bit more. There's two scenarios:
1. they've made one change, such as create a file, and want to make additional changes
--- implication: they need to be on the same branch as their initial change
2. they've made no previous changes on a branch, and just want to edit an article
--- implication: they just have to be on the `develop` main branch and click `edit`. Nothing special.

**Comment by Connoropolous** [comment]
@thedavidmeister re-review?
